piff/model.py

```diff
@@ -110,6 +110,26 @@ def fit(self, star):
         """
         raise NotImplemented("Derived classes must define the fit function")
 
+    def solve(self, star_list, logger=None):
+        """Solve for any model coefficients given some data.
+
+        :param star_list:   A list of Star instances to fit model parameters with
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        By default does NOTHING since you may have no hyperparameters to put in
+        """
+        pass
```

Comment:
I'm not sure I understand what this is supposed to do when it's not doing nothing.  Perhaps it will make more sense when coupled to your OpticalPSF classes and I can see a real world non-trivial example.